### PR TITLE
Editor: Floating menu using `ContextMenu`

### PR DESCRIPTION
**Pull request by barklund**
## Context

This PR converts the context menu to use flexbox and allows it to be used in both vertical and horizontal variants.

It then proceeds to use a horizontal context menu in the floating menu

## Relevant Technical Choices

* This touches the layout of context menu items and the menu itself, including using flexbox with gap rather than margin for spacing. It also modifies buttons, icons, and separators.
* This also refactor keyboard navigation in context menus to abstract out how you move between different types of elements.
* This does not (yet) include necessary fixes for submenus in horizontal menus.

## To-do

* This does not include work for regular text buttons, only icon buttons for now.
* This does not include the new types of elements needed, e.g. input elements and toggle buttons.
* Submenus for horizontal menus might need to be supported.
* Tooltips don't really work because it's white on white.

## User-facing changes

Compare the context menu before and after - to me it looks pretty pixel perfect:
| Focus | Before | After |
|-|-|-|
| Page side menu | Screen Shot 2022-02-01 at 19 38 26 | Screen Shot 2022-02-01 at 19 37 36 |
| Page side menu hover | Screen Shot 2022-02-01 at 19 38 29 | Screen Shot 2022-02-01 at 19 37 41 |
| Right-click menu | Screen Shot 2022-02-01 at 19 38 36 | Screen Shot 2022-02-01 at 19 37 53 |
| Right-click menu hover | Screen Shot 2022-02-01 at 19 38 39 | Screen Shot 2022-02-01 at 19 37 57 | 

The new floating menu now looks like this:

| Focus | Design |
|-|-|
| Regular | Screen Shot 2022-02-02 at 17 15 20 |
| Hover | Screen Shot 2022-02-02 at 17 15 24 |

## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Partially addresses #10112

### CODE REVIEW

File: packages/design-system/src/components/contextMenu/menu.js
Comment (barklund) [inline packages/design-system/src/components/contextMenu/menu.js:48]: This is where I moved to a flexbox layout with gaps rather than margins. That makes it a lot easier for switching directions (and works out of the box for RTL).

**Comment by barklund** [comment]
@samwhale, @BrittanyIRL, @littlemilkstudio: I'm not sure which of you is the right person to discuss this slight refactor of the context menu, but I of course only need input from one of you, so you don't all have to jump in here.

I will also be doing more stuff as I'm going to have to change some other things as well, so it would be good to know who to tag in for future PRs with changes to the context menu.

File: packages/design-system/src/components/contextMenu/menu.js
Comment (merapi) [inline packages/design-system/src/components/contextMenu/menu.js]: Should it be still `small` or [Shadow border radius](https://github.com/GoogleForCreators/web-stories-wp/blob/e00e50da7a02d956b993f4038ca1c15d46eda3ef/packages/design-system/src/components/contextMenu/styled.js#L42) should be changed to `medium`?

Otherwise:
image

**Comment by miina** [comment]
@barklund For some reason, the static menu with submenu is not displaying anymore in the Storybook:

Screenshot 2022-02-03 at 13 43 12

File: packages/design-system/src/components/contextMenu/menu.js
Comment (miina) [inline packages/design-system/src/components/contextMenu/menu.js:183]: (Also will depend on if the submenu opens to the top or to the bottom, e.g. if there's no room below.)

File: packages/design-system/src/components/contextMenu/components/button.js
Comment (miina) [inline packages/design-system/src/components/contextMenu/components/button.js:43]: Should it be `padding: hasNoPadding ? 0 : '2px 16px'`? Or is that the default?

File: packages/design-system/src/components/contextMenu/components/separator.js
Comment (miina) [inline packages/design-system/src/components/contextMenu/components/separator.js]:
```suggestion
        margin: 2px auto; /* there's an extra 6px vertical flexbox gap */
```

File: packages/design-system/src/components/contextMenu/components/separator.js
Comment (miina) [inline packages/design-system/src/components/contextMenu/components/separator.js]: What's the 31px? It's just a value from Figma? (Should it be 32 instead? Since we generally use *8 values, or even values)

**Review by miina — APPROVED** [review]
LGTM (pending the Storybook menu with submenu fix) but would be great to get a review from Pea pod as well.

File: packages/design-system/src/components/contextMenu/components/button.js
Comment (barklund) [reply packages/design-system/src/components/contextMenu/components/button.js:43]: The other padding is default, so no need to redefine it. This is just about removing padding or not.

File: packages/design-system/src/components/contextMenu/components/separator.js
Comment (barklund) [reply packages/design-system/src/components/contextMenu/components/separator.js]: Yes, I can change it to 32px, which is probably the intent, yes.

**Comment by barklund** [comment]
> @barklund For some reason, the static menu with submenu is not displaying anymore in the Storybook:

There wasn't anything there before either... Not sure what's up with that:

https://github.com/GoogleForCreators/web-stories-wp/blob/1bc0224f2b28f303918d8175d24f405b552e6fe2/packages/design-system/src/components/contextMenu/stories/index.js#L748-L750

**Comment by miina** [comment]
> There wasn't anything there before either... Not sure what's up with that:

Hum, it definitely was there when it was added first 🤔 
Will create an issue for it.

Issue: https://github.com/GoogleForCreators/web-stories-wp/issues/10454

**Review by BrittanyIRL — COMMENTED** [review]
Works for me! Marcin flagged the only thing I would have added - the border diff. Will approve when that is fixed.

**Comment by samwhale** [comment]
Thanks for tagging me on this! I've had the most context with the 'context menu', so I'd love to see any future work on it too!

I like the changes - it makes a lot of sense to convert the `ContextMenu` into a `FloatingMenu`. One concern I have is that the current `ContextMenu` is using a [menu](https://w3c.github.io/aria-practices/#menu) aria pattern. It seems like the floating menu could also be a menu/menu bar but wanted to get your take on it too.

If it's not, then we'll need to do some more work since all the components have specific `menu`/`menuitem` roles

File: packages/design-system/src/components/contextMenu/menu.js
Comment (barklund) [reply packages/design-system/src/components/contextMenu/menu.js]: Hm, I'm not seeing this? But perhaps I changed things after you reviewed:

Screen Shot 2022-02-04 at 10 12 12

The border-radius on the shadow is [already flexible depending on the menu orientation](https://github.com/GoogleForCreators/web-stories-wp/pull/10438/files#diff-e98500a7b8fdf0a25ebba0b1723fcbe6d5ca8d9afedb0773d1cd4bad29b9078bR42-R43).

Comment (BrittanyIRL) [reply packages/design-system/src/components/contextMenu/menu.js]: i just pulled down the latest and can confirm that this issue is resolved.

**Review by BrittanyIRL — APPROVED** [review]
Border radius issue is resolved ✅

File: packages/design-system/src/components/contextMenu/menu.js
Comment (merapi) [reply packages/design-system/src/components/contextMenu/menu.js]: Yep, fixed in https://github.com/GoogleForCreators/web-stories-wp/pull/10438/commits/87aa7bcc20470a917bcd1819f28bf1fd3db6c417

File: packages/design-system/src/components/contextMenu/stories/index.js
Comment (BrookeGraham) [inline packages/design-system/src/components/contextMenu/stories/index.js:408]: Hi! Just wanted to give you a heads up that `actions` are getting updated with the new Storybook controls. We now can define them within `argTypes` along with `args` at the top of the Storybook component. Everything has been upgraded so feel free to use the [new pattern](https://github.com/GoogleForCreators/web-stories-wp/pull/10274). The old `actions` have already been removed from the `Dashboard`, `WP-Dashboard` and `WP-Story-Editor` packages. The [Design-System](https://github.com/GoogleForCreators/web-stories-wp/pull/10520) and [Story-Editor](https://github.com/GoogleForCreators/web-stories-wp/pull/10499) packages will likely have the "old" `actions` removed later today or tomorrow.

**Pull request merged**
